### PR TITLE
NPT-535: Some Doc change related to Deprecate the use of maps to represent nexus links and children

### PR DESCRIPTION
**Pull request by LokeshvarKr**
[NPT-535] Some Documentation change related to Deprecate the use of maps to represent nexus links and children
And removing a duplicate test case.

### CODE REVIEW

File: compiler/example/datamodel/config/gns/service-group/service-group.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ type SvcGroup struct {
 	Description string
 	Color       string
 	// TODO support links which are not nexus nodes https://jira.eng.vmware.com/browse/NPT-112
-	//Services    map[string]core_v1.Service `nexus:"link"`
+	//Services    core_v1.Service `nexus:"links"`
```
Comment (xmen4xp) [inline compiler/example/datamodel/config/gns/service-group/service-group.go:14]: @LokeshvarKr please remove these 2 lines. They are no longer relevant.

**Review by xmen4xp — DISMISSED** [review]

File: compiler/pkg/parser/node_parser_test.go [hunk 2]
```diff
@@ -64,18 +64,6 @@ var _ = Describe("Node parser tests", func() {
 		Expect(fail).To(BeTrue())
 	})
 
-	It("should fail when nexus child or link fields is an array.", func() {
```
Comment (PavitraS) [inline compiler/pkg/parser/node_parser_test.go:67]: What caused the deletion of this test case?

File: compiler/example/datamodel/config/gns/service-group/service-group.go (hunk 1 above)
Comment (LokeshvarKr) [reply compiler/example/datamodel/config/gns/service-group/service-group.go:14]: Sure @xmen4xp

File: compiler/pkg/parser/node_parser_test.go (hunk 2 above)
Comment (LokeshvarKr) [reply compiler/pkg/parser/node_parser_test.go:67]: test case was already there 
https://github.com/vmware-tanzu/graph-framework-for-microservices/blob/795eaa238ec218a7a6a0b695f206a05c39dd1fac/compiler/pkg/parser/node_parser_test.go#L67

and I also I added one similar test case 

https://github.com/vmware-tanzu/graph-framework-for-microservices/blob/795eaa238ec218a7a6a0b695f206a05c39dd1fac/compiler/pkg/parser/node_parser_test.go#L91

Comment (LokeshvarKr) [reply compiler/pkg/parser/node_parser_test.go:67]: @PavitraS  PTAL

**Review by PavitraS — APPROVED** [review]

**Pull request merged**